syntax-test_dev.py

```diff
@@ -236,11 +237,30 @@ def on_selection_modified_async(self):
             col_end += 1
 
         region = sublime.Region(line.begin() + col_start, line.begin() + col_end)
-        self.view.add_regions('current_syntax_test', [region], 'text', '', sublime.DRAW_NO_FILL)
-        # originally, this was `sublime.DRAW_NO_OUTLINE | sublime.DRAW_SOLID_UNDERLINE
-        # | sublime.DRAW_NO_FILL`
-        # but underlines aren't drawn on spaces
-        # see https://github.com/SublimeTextIssues/Core/issues/137
+
+        prefs = sublime.load_settings('PackageDev.sublime-settings')
+        scope = prefs.get('scope_for_syntax_test_highlight', 'text')
+        styles = prefs.get('styles_for_syntax_test_highlight', ['DRAW_NO_FILL'])
+        style_flags = 0
+        # # the following available add_region styles are taken from the API documentation:
+        # # http://www.sublimetext.com/docs/3/api_reference.html#sublime.View
+        # # unfortunately, the `sublime` module doesn't encapsulate them for easy reference
+        # for style in styles:
+        #     if style in [
+        #         'DRAW_EMPTY', 'HIDE_ON_MINIMAP', 'DRAW_EMPTY_AS_OVERWRITE', 'DRAW_NO_FILL',
+        #         'DRAW_NO_OUTLINE', 'DRAW_SOLID_UNDERLINE', 'DRAW_STIPPLED_UNDERLINE',
+        #         'DRAW_SQUIGGLY_UNDERLINE', 'HIDDEN', 'PERSISTENT'
+        #     ]:
+        #         style_flags |= getattr(sublime, style)
+        for style in styles:
+            # check the style matches the casing and naming used for constants in `sublime.py`
```

Comment:
I prefer the explicit check here since the module contains other constants that would not make sense in this context. I know that you hardly ever encounter a situation where a user inserts this in their settings, but I like to be on the safe side when evaluating user input, always.